lisa/sut_orchestrator/baremetal/source.py

```diff
@@ -61,25 +89,31 @@ def download(self, timeout: int = 600) -> List[Path]:
             pipeline_name=pipeline_name,
             timeout=timeout,
         )
-        artifacts_path_raw = artifacts_path
-        artifacts_path = []
-        for artifact in artifacts:
-            pattern = re.compile(rf".*{artifact.artifact_name}.*")
-            for artifact_path in artifacts_path_raw:
-                if pattern.match(str(artifact_path)):
-                    self._log.info(f"Artifact downloaded to {str(artifact_path)}")
-                    if artifact.extract:
-                        source_path = self.extract(artifact_path)
-                        artifacts_path.append(Path(source_path))
-                    else:
-                        artifacts_path.append(artifact_path)
-        return artifacts_path
-
-    def extract(self, artifact_path: Path) -> str:
-        file_extension = artifact_path.suffix
-        if file_extension == ".zip":
-            with zipfile.ZipFile(str(artifact_path), "r") as zip_ref:
-                zip_ref.extractall(str(artifact_path.parent))
-        source_path = os.path.splitext(str(artifact_path))[0]
-        self._log.info(f"Artifact extracted to {str(source_path)}")
-        return source_path
+
+        return _process_extract(artifacts, artifacts_path)
+
+
+class LocalSource(Source):
+    def __init__(self, runbook: LocalSourceSchema) -> None:
+        super().__init__(runbook)
+        self.local_runbook: LocalSourceSchema = runbook
+        self._log = get_logger("local", self.__class__.__name__)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "local"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return LocalSourceSchema
+
+    def download(self, timeout: int = 600) -> List[Path]:
+        local_artifacts_path: List[Path] = []
+
+        for artifact in self.local_runbook.artifacts:
+            local_artifacts_path.append(Path(artifact.artifact_name))
+
+        return _process_extract(
```

Comment:
Update the SMBSource to use the same logic.

Reply:
The change is already done, thanks